### ISSUE TITLE
Forbid repeating conditions in the one condition block

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1004]
# Rule request

## Thesis

In examples below, `x`, `y` and `z` are any Python expressions. For example, `x and x` can be `func(a, b, c=3) and func(a, b, c=3)`.

Bad:

```python
if x and x: ...
if x or x: ...
if x and not x: ...
if x or not x: ...
```

Good:

```
if x: ...
if not x: ...
```

If we're brave enough, we can after this rule care a little bit about conditions simplification:

```
x or x and y == x or y
x and y or y and z == y and (x or z)
```

## Reasoning

In most of cases it's a logical mistake. Every condition above can be reduced to `x`, `True` or `False`, and usually, it's not what you expect.

**Comment by sobolevn** [comment]
It is partially supported right now:

```python
ex.py

  1:4      WPS408 Found duplicate logical condition
  if x and x: ...
     ^

  2:4      WPS408 Found duplicate logical condition
  if x or x: ...
```

We need to improve the check for this rule.
